src/rtObject.cpp

```diff
@@ -56,13 +56,14 @@ rtError rtEmit::setListener(const char* eventName, rtIFunction* f)
     e.isProp = true;
     e.markForDelete = false;
     e.fnHash = f->hash();
-    mEntries.push_back(e);      
+    e.emitOnce = false;
+    mEntries.push_back(e);
   }
   
   return RT_OK;
 }
 
-rtError rtEmit::addListener(const char* eventName, rtIFunction* f)
+rtError rtEmit::addListener(const char* eventName, rtIFunction* f, bool emitOnce)
```

Comment:
In what cases do we only want to emit once?

Reply:
rtHttp* classes should work like node's EventEmitter which has 'once' as well as 'on'.